### PR TITLE
MCO-687: Fix metrics e2e test

### PR DESCRIPTION
**Pull request by cheesesashimi**
**- What I did**

The TestMetrics e2e test is not correctly cleaning up the MachineConfigs and MachineConfigPools it creates. This means that other e2e tests which run after this e2e test can falsely fail or become flaky.

What's happening is this:

1. The target node is removed from the ephemeral MachineConfigPool by unlabeling it.
2. A race condition occurs when we call WaitForPoolComplete because technically, the pool is updated at this point since it has not yet picked up the unlabelling event from the target node.
3. We delete the ephemeral MachineConfigPool, which deletes the rendered MachineConfigs that belong to it.
4. The node starts the update process, but cannot find the rendered MachineConfigs for the ephemeral pool since they were deleted. The MCD degrades at this point and blocks the worker MachineConfigPool.

What should be happening is this:

1. The target node is removed from the ephemeral MachineConfigPool by unlabeling it.
2. Wait until the target node completes the switch back to the worker pool.
3. Delete the ephemeral MachineConfigPool that was created for the test.
4. Delete any MachineConfigs assigned to that ephemeral MachineConfigPool.

This PR also includes some opportunistic refactoring and de-duplication of the helper functions used for this and other e2e tests.

**- How to verify it**

1. Provision an OpenShift cluster.
2. Run the `TestMetrics` test: `$ go test -run="TestMetrics" -count -v .`
3. Check that the `worker` MachineConfigPool has not degraded after the test is complete.

**- Description for the changelog**
TestMetrics e2e test should clean up in the correct order

### CODE REVIEW

File: test/helpers/utils.go [hunk 1]
```diff
@@ -491,14 +482,18 @@ func CreateMCP(t *testing.T, cs *framework.ClientSet, mcpName string) func() {
...
-		if !errors.IsAlreadyExists(err) {
-			require.Nil(t, err)
-		}
+	if err == nil {
+		t.Logf("Created MachineConfigPool %q", mcpName)
+	}
+	if errors.IsAlreadyExists(err) {
+		t.Logf("Reusing existing MachineConfigPool %q", mcpName)
+	} else {
+		require.NoError(t, err)
```
Comment (cdoern) [inline test/helpers/utils.go:491]: well we know err exists here, sould this be `t.Fatalf`?

**Review by cdoern — APPROVED** [review]
Looks good, thanks for cleaning this functionality up.

File: test/helpers/utils.go (hunk 1 above)
Comment (cheesesashimi) [reply test/helpers/utils.go]: Fixed!

**Pull request merged**